tap_shopify/__init__.py

```diff
@@ -14,14 +14,35 @@
 from tap_shopify.context import Context
 import tap_shopify.streams # Load stream objects into Context
 
-REQUIRED_CONFIG_KEYS = ["shop", "api_key"]
+REQUIRED_CONFIG_KEYS = ["shop"]
 LOGGER = singer.get_logger()
 
 def initialize_shopify_client():
-    api_key = Context.config['api_key']
     shop = Context.config['shop']
-    session = shopify.Session(shop, api_key)
-    shopify.ShopifyResource.activate_session(session)
+
+    if 'api_key' in Context.config:
+        LOGGER.info("Using public app token based authentication from config[\"api_key\"]")
+        session = shopify.Session(shop, Context.config['api_key'])
+        shopify.ShopifyResource.activate_session(session)
+    elif 'private_app_api_key' in Context.config and 'private_app_password' in Context.config:
+        LOGGER.info(
+            "Using private app basic authentication from config[\"private_app_api_key\"] "
+            "and config[\"private_app_password\"]"
+        )
+
+        api_key = Context.config['private_app_api_key']
+        password = Context.config['private_app_password']
+        shop_url = 'https://{}:{}@{}/admin'.format(api_key, password, shop)
+
+        shopify.ShopifyResource.set_site(shop_url)
+    else:
+        raise Exception(
+            "Config is missing Shopify auth information." +
+            "Either pass the `api_key` option for a public app authenticated using an" +
+            " access_token via the Shopify OAuth flow, or pass the `private_app_api_key`" +
+            " and `private_app_password` config options for a private app."
+        )
```

Comment:
This worked with `pylint` locally and is how I like to do multiline strings in python
```
        raise Exception(
            ("Config is missing Shopify auth information. Either pass the `api_key` option for a "
             "public app authenticated using an access_token via the Shopify OAuth flow, or pass "
             "the `private_app_api_key` and `private_app_password` config options for a private "
             "app."))
```